File: backend/src/routers/authRouter.ts
import { publicProcedure, router } from "../trpc.js";
import { z } from "zod";

export const authRouter = router({
  register: publicProcedure
    .input(
      z.object({
        password: z.string(),
        email: z.string().email(),
      }),
    )
    .mutation(async ({ input: { email, password } }) => {}),
  login: publicProcedure
    .input(
      z.object({
        email: z.string(),
        password: z.string(),
      }),
    )
    .mutation(async ({ input: { email, password } }) => {}),
});

File: backend/src/modules/auth/auth_router.py
from fastapi import APIRouter
from pydantic import BaseModel, SecretStr

from backend.src.modules.auth.user import User

auth_router = APIRouter()

class LoginUserDTO(BaseModel):
    email: str
    password: SecretStr


@auth_router.post('/register')
def register(dto: LoginUserDTO):
    user = User.create_from(email=dto.email, password=dto.password.get_secret_value(),)

@auth_router.post('/login')
def login(dto: LoginUserDTO):
    user = User.create_from(email=dto.email, password=dto.password.get_secret_value(),)
